definitions/procedures/pulp/cleanup_old_metadata_files.rb

```diff
@@ -0,0 +1,74 @@
+require 'find'
+require 'rexml/document'
+
+module Procedures::Pulp
+  class CleanupOldMetadataFiles < ForemanMaintain::Procedure
+    metadata do
+      description 'Cleanup old and unneeded yum metadata files from /var/lib/pulp/'
+      confine do
+        check_max_version('katello-common', '4.0')
+      end
+      param :remove_files, 'If true, will actually delete files, otherwise will print them out.'
+    end
+
+    PULP_METADATA_DIR = '/var/lib/pulp/published/yum/master/yum_distributor/'.freeze
+    REPOMD_FILE = 'repomd.xml'.freeze
+
+    def run
+      return unless File.directory?(PULP_METADATA_DIR)
+      found = []
+
+      puts 'Warning: This command may cause reduced performance related to content operations.'
+      message = 'Locating repository metadata (this may take a while)'
+      with_spinner(message) do |spinner|
+        Find.find(PULP_METADATA_DIR) do |path|
+          next if File.basename(path) != REPOMD_FILE
+          found << path
+          spinner.update("#{message}: Found #{found.count} repos.")
+        end
+      end
+
+      found.each do |repo_md_path|
+        handle(repo_md_path, @remove_files)
+      end
+    end
+
+    def handle(repo_md_path, remove_files)
+      base_path = File.dirname(repo_md_path)
+      to_remove = list_existing_files(repo_md_path) - list_repomd_files(repo_md_path)
+
+      if to_remove.empty?
+        "Skipping #{base_path}, no files to remove."
+      elsif remove_files
+        puts "Removing #{to_remove.count} files from #{base_path}"
+        to_remove.each { |file| File.delete(File.join(base_path, file)) }
+      else
+        puts '============================================='
```

Comment:
Is this puts required? if yes then can this be of same size of spinner,
~~~
# bin/foreman-maintain content cleanup-repository-metadata
Running Remove old leftover repository metadata
================================================================================
Cleanup old and unneeded yum metadata files from /var/lib/pulp/: 
Warning: This command may cause reduced performance related to content operations.
/ Locating repository metadata (this may take a while): Found 1 repos.          
=============================================
For /var/lib/pulp/published/yum/master/yum_distributor/f168cf9f-72e2-4bc3-a02a-f0633e2c937e/1625844163.93/repodata would remove, but --remove-files was not specified:
~~~

If you think puts is good to have then can we also add this in `elsif remove_files` block,
~~~
      elsif remove_files
        puts '================================================================================'
        puts "Removing #{to_remove.count} files from #{base_path}"
        to_remove.each { |file| File.delete(File.join(base_path, file)) }
~~~

Reply:
I added it to elsif too.  I don't think its necessary but is a nice to have to split up the different sections of output.  I also extended it to make it longer 👍🏻

Reply:
updated to 81 chars